src/main/java/gregtech/api/fluids/MaterialFluidBlock.java

```diff
@@ -0,0 +1,79 @@
+package gregtech.api.fluids;
+
+import gregtech.api.GTValues;
+import gregtech.api.unification.material.Material;
+import gregtech.api.unification.material.info.MaterialFlags;
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockFire;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.entity.Entity;
+import net.minecraft.init.Blocks;
+import net.minecraft.util.BlockRenderLayer;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.IBlockAccess;
+import net.minecraft.world.World;
+import net.minecraftforge.fluids.BlockFluidClassic;
+import net.minecraftforge.fluids.Fluid;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public class MaterialFluidBlock extends BlockFluidClassic {
+
+    private final Material gtMaterial;
+    private final boolean isFlammable;
+    private final boolean isExplosive;
+    private final boolean isSticky;
+
+    public MaterialFluidBlock(@Nonnull Fluid fluid, @Nonnull net.minecraft.block.material.Material material, @Nonnull Material gtMaterial) {
```

Comment:
Why is the second parameter here the Minecraft Material, when all that is ever passed into this method is `GTFluidMaterial`? Since GTFluidMaterial extends MC's Material, having that as a parameter and passing it to `super` would work fine.